test/common/schemaDynamicTest.js

```diff
@@ -0,0 +1,119 @@
+'use strict';
+
+var assign = require('lodash').assign;
+var chai = require('chai');
+var expect = require('chai').expect;
+require('it-each')({ testPerIteration: true });
+
+var typesRepresentatives = require('./typesRepresentatives');
+
+var arrays = typesRepresentatives.arrays;
+var booleans = typesRepresentatives.booleans;
+var integers = typesRepresentatives.integers;
+var numbers = typesRepresentatives.numbers;
+var objects = typesRepresentatives.objects;
+var others = typesRepresentatives.others;
+var strings = typesRepresentatives.strings;
+
+var self;
+
+function SchemaDynamicTest (config) {
+
+	this.customInput = config.customInput;
+	this.customArgumentAssertion = config.customArgumentAssertion;
+	this.customPropertyAssertion = config.customPropertyAssertion;
+	this.customRequiredPropertiesAssertion = config.customRequiredPropertiesAssertion;
+
+	self = this;
+
+	self.schema = {
+		shouldFailAgainst: {
+			nonArray: {
+				arguments: self.testArgument.bind(self, 'array', booleans.concat(integers).concat(numbers).concat(objects).concat(others).concat(strings)),
```

Comment:
It'd be more readable if we have a superset of all types, and we remove the entries which are valid rather than concatenating all invalid types each time. Like,
`arguments: _.difference(allTypes, arrays)`